agent/banner/banner.go

```diff
@@ -167,6 +170,8 @@ func (w *bannerResponseWriter) getBanner(favIconLink string) ([]byte, error) {
 }
 
 func (w *bannerResponseWriter) WriteHeader(statusCode int) {
+	w.metricHandler.WriteResponseCodeMetric(statusCode)
```

Comment:
This seems like it's double counting the response metric since this is also called on line 214 below (you'll call writeheader, record the status code, then return and immediately record it again). In addition, if we choose to keep this response code call in this function, we should move it below the w.wroteHeader check for the same reason.

Reply:
Ah nice catch. I think it's better to remove it Write() and keep it in WriteHeader(), since there may be places where only WriteHeader() is called.

Reply:
I think we actually need to keep in WriteHeader but *not* Write, since as per Go documentation you can call writeheader without calling write, but if you call write without calling writeheader it will automatically call writeheader(200). So we should remove the call in Write, and move this call to below the if w.wroteHeader check.

Reply:
Done